web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx

```diff
@@ -1,45 +1,92 @@
-import { useEffect, useState } from "react";
-import { ToastType, useToast } from "@humansignal/ui";
 import { format } from "date-fns";
 import styles from "./MembershipInfo.module.scss";
+import { useQuery } from "@tanstack/react-query";
 
 /**
  * FIXME: This is legacy imports. We're not supposed to use such statements
  * each one of these eventually has to be migrated to core/ui
  */
 import { useCurrentUser } from "/apps/labelstudio/src/providers/CurrentUser";
-import { useAPI } from "/apps/labelstudio/src/providers/ApiProvider";
+import { API } from "apps/labelstudio/src/providers/ApiProvider";
+import { useMemo } from "react";
 
 export const MembershipInfo = () => {
-  const api = useAPI();
-  const toast = useToast();
   const { user } = useCurrentUser();
-  const [registrationDate, setRegistrationDate] = useState<string | null>(null);
-  const [annotationsCount, setAnnotationsCount] = useState<number | null>(null);
-  const [projectsContributedTo, setProjectsContributedTo] = useState<number | null>(null);
-
-  useEffect(() => {
-    if (!user) return;
-    api
-      .callApi("userMemberships", {
-        params: {
-          pk: user.active_organization,
-          userPk: user.id,
-        },
-      })
-      .then((response: any) => {
-        if (response?.$meta?.ok) {
-          setRegistrationDate(format(new Date(response?.created_at), "dd MMM yyyy, KK:mm a"));
-          setAnnotationsCount(response?.annotations_count);
-          setProjectsContributedTo(response?.contributed_projects_count);
-        } else {
-          toast.show({
-            message: "Failed to fetch membership info",
-            type: ToastType.error,
-          });
-        }
+  const dateJoined = useMemo(() => {
+    if (!user?.date_joined) return null;
+    return format(new Date(user?.date_joined), "dd MMM yyyy, KK:mm a");
+  }, [user?.date_joined]);
+  console.log("====================");
+
+  const membership = useQuery({
+    queryKey: [user?.active_organization, user?.id, "user-membership"],
+    async queryFn() {
+      if (!user) return {};
+      console.log("loading membership");
+      const response = await API.invoke("userMemberships", {
+        pk: user.active_organization,
+        userPk: user.id,
+      });
+
+      console.log("user", response);
+
+      const registrationDate = format(new Date(response?.created_at), "dd MMM yyyy, KK:mm a");
+      const annotationCount = response?.annotations_count;
+      const contributions = response?.contributed_projects_count;
+
+      return {
+        registrationDate,
+        annotationCount,
+        contributions,
+      };
+    },
+  });
+
+  const organization = useQuery({
+    queryKey: ["organization", user?.active_organization],
+    async queryFn() {
+      if (!user) return null;
+      if (!window.APP_SETTINGS.billing) return null;
```

Comment:
i've noticed that when doing tests this can cause issues - making it null safe avoids that
```suggestion
      if (!window?.APP_SETTINGS?.billing) return null;
```